src/abstractdataframe/join.jl

```diff
@@ -417,8 +417,9 @@ function _join(df1::AbstractDataFrame, df2::AbstractDataFrame;
 
     if indicator !== nothing
         refs = left_indicator + right_indicator
-        pool = CategoricalPool{String,UInt8}(["left_only", "right_only", "both"])
-        indicatorcol = CategoricalArray{String,1}(refs, pool)
+        refs_short, invpool, pool = PooledArrays._label(["left_only", "right_only", "both"], String)
```

Comment:
How about creating the dicts manually to avoid relying on an internal function?

Reply:
I was considering this. What I do checks if what PooledArrays.jl produces and uses is consistent (i.e. if PooledArrays.jl changes its internal design we will get an error here and this is good; if we create dicts manually and PooledArrays.jl is changed we might silently start producing wrong results).

What do you think we should do?

Reply:
AFAICT passing dicts and refs is part of the public API, so we're safe.

Reply:
OK - I will rewrite it.

Reply:
done